lib/src/localUpdater/iso_storage_updater.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:async';
+import 'dart:isolate';
+
+import 'package:matrix_sdk/src/localUpdater/sink_response.dart';
+
+import '../../../matrix_sdk.dart';
+import '../updater/isolated/utils.dart';
+import '../util/logger.dart';
+import 'instruction.dart';
+import 'local_updater.dart';
+
+abstract class IsolateStorageUpdater {
+  static Future<void> run(IsolateTransferModel transferModel) async {
+    final message = transferModel.message;
+    Log.setLogger(transferModel.loggerVariant);
+    final receivePort = ReceivePort();
+    final messageStream = receivePort.asBroadcastStream();
+    final sendPort = message as SendPort;
+
+    await runZonedGuarded(
+      () async {
+        LocalUpdater? updater;
+        StreamSubscription? subscription;
+        StreamSubscription? updateSubscription;
+        StreamSubscription<Room>? roomSinkSubscription;
+
+        sendPort.send(receivePort.sendPort);
+
+        subscription = messageStream.listen((message) async {
+          //first sink, after [receivePort.sendPort] send
+          if (message is IsoStorageUpdaterArgs) {
+            updater = LocalUpdater(
+              storeLocation: message.storeLocation,
+              isIsolated: false,
+            );
+            await updater?.init();
+            await updater?.ensureReady();
+            updateSubscription = updater?.userUpdates.listen(sendPort.send);
+            sendPort.send(IsolateStorageSyncerInitialized());
+            //On start sink
+          } else if (message is IsolateStorageStartSyncInstruction) {
+            updater?.initSinkStorage();
```

Comment:
minor. looks like typo. should be 'sync' instead of 'sink'

Reply:
Renamed all sink occurrence into sync